include/boost/http_proto/detail/impl/array_of_buffers.hpp

```diff
@@ -105,6 +115,18 @@ consume(std::size_t n)
         detail::throw_logic_error();
 }
 
+template<bool isConst>
+void
+array_of_buffers<isConst>::
+reset(std::size_t n)
```

Comment:
There is danger if this function is needed... array of buffers is for type-erasing the user's sequence. `reset` implies some kind of re-use which may not be appropriate.

Reply:
Yup, the whole goal is reuse here.

I realized, we were only half-using `out_` before so I updated it to be reusable and that's the core of the refactor.

Because of this, `serializer::consume()` is much more straight-forward as well, it's just `out_.consume(n)` calls now.

This later improves locality of reasoning in other spots in the code as well.